### PR TITLE
Update release-3.4 and 3.5 Changelog for go bump to 1.20.12

### PR DESCRIPTION
**Pull request by jonasrdl**
https://github.com/etcd-io/etcd/issues/17049

https://github.com/etcd-io/etcd/pull/17076#issuecomment-1843614913

### CODE REVIEW

**Comment by jonasrdl** [comment]
If this is fine, I'll also raise a PR with changelog changes for 3.5

File: CHANGELOG/CHANGELOG-3.4.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 Previous change logs can be found at [CHANGELOG-3.3](https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.3.md).
 
+## v3.4.29
```
Comment (fuweid) [inline CHANGELOG/CHANGELOG-3.4.md:5]: It should be appended `(tbd)`

**Review by fuweid — APPROVED** [review]
LGTM

**Review by jmhbnz — APPROVED** [review]
Hey @jonasrdl - This looks good thanks, feel free to just update both CHANGELOGs in this pr to save time.

**Comment by jonasrdl** [comment]
@jmhbnz sure, I also pushed 3.5.
I noticed, `v3.5.11` got released 11 hours ago, but in the changelog is still `(tbd)`. Should I fix that also in this PR?

**Review by fuweid — APPROVED** [review]
LGTM

File: CHANGELOG/CHANGELOG-3.5.md [hunk 2]
```diff
@@ -2,9 +2,14 @@
 
 Previous change logs can be found at [CHANGELOG-3.4](https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.4.md).
 
+## v3.5.12 (tbd)
+
+## Dependencies
+- Compile binaries using [go 1.20.12](https://github.com/etcd-io/etcd/pull/17077)
```
Comment (serathius) [inline CHANGELOG/CHANGELOG-3.5.md:8]: Compiling with go1.20.12 went into v3.5.12

Comment (serathius) [reply CHANGELOG/CHANGELOG-3.5.md:8]: https://github.com/etcd-io/etcd/commits/v3.5.11

Comment (jonasrdl) [reply CHANGELOG/CHANGELOG-3.5.md]: Sorry, mixed things up. Should be fine now

**Review by ahrtr — APPROVED** [review]

**Review by serathius — APPROVED** [review]

**Pull request merged**